src/wallet/walletdb.cpp

```diff
@@ -298,6 +301,40 @@ bool WalletBatch::EraseLockedUTXO(const COutPoint& output)
     return EraseIC(std::make_pair(DBKeys::LOCKED_UTXO, std::make_pair(output.hash, output.n)));
 }
 
+bool WalletBatch::WriteHDKey(const CExtKey& extkey)
+{
+    std::vector<unsigned char> xpub(BIP32_EXTKEY_SIZE);
+    extkey.Neuter().Encode(xpub.data());
+
+    CPrivKey privkey = extkey.key.GetPrivKey();
+
+    return WriteIC(std::make_pair(DBKeys::HDKEY, xpub), std::make_pair(privkey, PrivKeyChecksum(privkey, xpub)), false);
```

Comment:
I know that this is a limitation of our current db model. But, the amount of db records containing this private key is, at least, unorthodox.

We are writing it 9 times under different db keys; one for the HDKey, then one time per descriptor.

Reply:
I do wonder why the default wallet stores the master extended private key in the private descriptor even though the key itself such as what you see in listdecriptors (not true) is a private child derivation.  It would be helpful for privacy if it did not do that. Ie: being able to spend one BIP32 account, allows being able to spend them all in the current default descriptors.

But if the master extended private key is going to be stored anyways now (does it need to be?) then no gain is possible. Seems like just the public masterkey could be stored or just the child extended public key for multisig.

Reply:
@furszy it would be better to address that now than to introduce yet another migration later.

@BenWestgate since unlocking a wallet decrypts everything, I don't think that matters. We could however make export safer by not including the root in the RPC output (and have some other way to obtain the seed), but that's unrelated to this PR.